vlad_task3/src/ServerProxyObject.cpp

```diff
@@ -0,0 +1,81 @@
+#include <cstdint>
+#include <ServerProxyObject.hpp>
+#include <boost/format.hpp>
+#include <Common.hpp>
+
+using namespace common;
+
+namespace server
+{
+    ServerProxyObject::ServerProxyObject(DynamicArray *arr) : ProxyBase(boost::interprocess::create_only), arr(arr)
+    {
+    }
+
+    void ServerProxyObject::run()
+    {
+        uint8_t command;
+        uint64_t recvd;
+        uint32_t priority;
+        while (active)
+        {
+            mq->receive(&command, sizeof(int), recvd, priority);
+            switch (command)
+            {
+            case EXIT:
+            {
+                active = false;
+            }
+            break;
+            case ADD:
+            {
+                {
+                    boost::interprocess::scoped_lock<boost::interprocess::named_mutex> lk(*mutex);
+                    arr->addEntry(getValue());
+                    setStatus(SUCCESS);
+                }
+                cond->notify_all();
```

Comment:
The condition notify occur in every case, maybe you can write it after the switch that should include also a default for wrong commands.